File: Python/Structures/Stack.py
class Node:
    def __init__(self, data: int) -> None:
        self.data = data
        self.next = None


class Stack:
    def __init__(self) -> None:
        self.top = None
        self.size = 0

    def push(self, data: int) -> None:
        node = Node(data)
        node.next = self.top
        self.top = node
        self.size += 1

    def pop(self) -> None:
        if self.is_empty():
            raise Exception("Stack is already")
        self.top = self.top.next
        self.size -= 1

    def display(self) -> None:
        ptr = self.top
        while ptr.next != None:
            print(ptr.data)
            ptr = ptr.next

    def is_empty(self) -> bool:
        return self.size == 0


if __name__ == '__main__':
    s = Stack()
    for i in range(0, 9):
        s.push(i)

    s.pop()
    s.display()

File: Swift/Structures/Stack.swift
final class Node{
    var data:Int;
    var next:Node? = nil;

    init(m_data:Int) {data = m_data;}
    // init(){}
}

class Stack{
    private var front:Node? = nil;
    private var back:Node? = nil;
    private var m_size:Int = 0;

    func push(data:Int){
        var node:Node = Node(data);
        if isEmpty(){
            self.front = node;
            self.back = node;
        }
        else{
            node.next = self.front;
            self.front = node;
        }
        self.m_size += 1;
    }

    func pop(){ // how GC is managed in swift?
        if isEmpty(){
            return;
        }
        self.front = self.front.next;
        self.m_size -= 1;
    }

    func display(){
        var ptr:Node? = self.front;
        while(ptr.next != nil){
            print(" \(ptr.data)");
            ptr = ptr.next;
        }
    }

    func isEmpty()->Bool{return head == nil;}
}